lib/ocpp/v201/smart_charging.cpp

```diff
@@ -102,8 +121,34 @@ ProfileValidationResultEnum SmartChargingHandler::validate_profile_schedules(con
     return ProfileValidationResultEnum::Valid;
 }
 
-void SmartChargingHandler::add_profile(const ChargingProfile& profile) {
-    charging_profiles.push_back(profile);
+void SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
+    if (STATION_WIDE_ID == evse_id) {
+        station_wide_charging_profiles.push_back(profile);
+    } else {
+        charging_profiles.push_back(EvseProfile{evse_id = evse_id, profile = profile});
+    }
+}
+
+std::vector<ChargingProfile> SmartChargingHandler::get_evse_specific_tx_default_profiles() const {
+    std::vector<ChargingProfile> evse_specific_tx_default_profiles;
+    for (auto iter = charging_profiles.begin(); iter != charging_profiles.end(); ++iter) {
```

Comment:
If it is possible I would suggest to use the range based for loop here and in the other for loops. More readable and less error prone.

Reply:
Done.

Reply:
Looks quite a lot cleaner :+1: